luigi/scheduler.py

```diff
@@ -1454,6 +1456,23 @@ def get_task_status_message(self, task_id):
         else:
             return {"taskId": task_id, "statusMessage": ""}
 
+    @rpc_method()
+    def set_task_progress_percentage(self, task_id, progress_percentage):
+        if self._state.has_task(task_id):
+            task = self._state.get_task(task_id)
+            task.progress_percentage = progress_percentage
+            if task.status == RUNNING and task.batch_id is not None:
+                for batch_task in self._state.get_batch_running_tasks(task.batch_id):
+                    batch_task.progress_percentage = progress_percentage
+
+    @rpc_method()
+    def get_task_progress_percentage(self, task_id):
+        if self._state.has_task(task_id):
+            task = self._state.get_task(task_id)
+            return {"taskId": task_id, "progressPercentage": task.progress_percentage}
+        else:
+            return {"taskId": task_id, "progressPercentage": None}
```

Comment:
Wait a second. Why don't we return `None`/`{}` here or some sort of failure indication, since the task id didn't exist right?